src/microbe_stage/IEngulfable.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System.Collections.Generic;
+
+/// <summary>
+///   Objects that can be engulfed by a microbe.
+/// </summary>
+[UseThriveSerializer]
+public interface IEngulfable : IGraphicalEntity
+{
+    /// <summary>
+    ///   The size of this engulfable object based on microbe hex count.
+    /// </summary>
+    public float Size { get; }
+
+    public float Radius { get; }
+
+    public EntityReference<Microbe> HostileEngulfer { get; }
+
+    /// <summary>
+    ///   The current step of phagocytosis process this engulfable is currently in.
+    /// </summary>
+    public PhagocytosisPhase PhagocytizedStep { get; set; }
+
+    /// <summary>
+    ///   What specific enzyme needed to digest (break down) this engulfable. If null default is used (lipase).
+    /// </summary>
+    public Enzyme? RequisiteEnzymeToDigest { get; }
+
+    public CompoundBag? Compounds { get; }
```

Comment:
Why is this nullable? Is there some place where null needs to be returned? I haven't seen all of the code here yet, but it might be overall simpler design for whatever place that has no compounds, to just return a "new CompoundBag", assuming this wouldn't get called too much, and there is no need to distinguish between things that are empty or can't have any compounds. If there is a need to distinguish those, then it's natural to have the nullable compound bag here.

Reply:
FloatingChunk has CompoundBag already nullable. I'm merely adapting this to it.

Reply:
But would it be too bad to modify floating chunk to return an empty compound bag when it is asked for the compounds when it doesn't have any? This interface is more widely used that the floating chunk so the design should consider what makes sense for this interface, and then adapt floating chunk if needed to work with that.

Reply:
Did that.